src/util/std_expr.h

```diff
@@ -3684,39 +3672,34 @@ inline void validate_expr(const array_update_exprt &value)
 
 /*! \brief Extract member of struct or union
 */
-class member_exprt:public exprt
+class member_exprt:public unary_exprt
 {
 public:
-  explicit member_exprt(const exprt &op):exprt(ID_member)
+  explicit member_exprt(const exprt &op):unary_exprt(ID_member, op)
   {
-    copy_to_operands(op);
   }
 
-  explicit member_exprt(const typet &_type):exprt(ID_member, _type)
+  explicit member_exprt(const typet &_type):unary_exprt(ID_member, _type)
   {
-    operands().resize(1);
   }
 
   member_exprt(const exprt &op, const irep_idt &component_name):
-    exprt(ID_member)
+    unary_exprt(ID_member, op)
```

Comment:
Yields the wrong type. I think this should be forbidden as well.

Reply:
Yes.